Basic benchmarking of solr indexing on nodes.

tylercloke opened the pull request:
To test out the benchmarking I have so far:

```
ssh into running chef server
git clone this
git checkout tc/benchmark
cp /etc/opscode/pivotal.rb .chef/knife-in-guest.rb
mv setup.yml.example setup.yml and change any ec settings
./setup-ec.sh (generates data on the chef-server)
time ./benchmark.sh
```

Currently its only benchmarking updating node attributes. Look in `testdata/created-objects.yml` after `./setup-ec.sh` to see what data is generated.

To add benchmarking, check out `scripts/ec/benchmark/solr.rb` and `benchmark.sh`, but basically, the latter is iterating through the admin user knife files that are generated (there is admin user / knife file per org generated).

If you wanna start from scratch with fresh data, `rm -rf testdata/*` and re-run `./setup-ec.sh` and `time ./benchmark.sh` (we should make this better).

tylercloke committed:
Basic benchmarking of solr indexing on nodes.

stevendanna commented on scripts/ec/benchmark/solr.rb:
```diff
@@ -0,0 +1,12 @@
+org = Chef::Config['chef_server_url'].split("/")[-1]
+nodes = created_objects["orgs"][org]["nodes"]
+
+nodes.each do |node|
```
I think we will probably want to test along to axis of variation:
- The number of concurrent writers
- The size of the objects being written

stevendanna committed:
Use set -e to ensure we easily see the first failure

stevendanna committed:
Add concurrency via threaded workers

Since we want to eventually test how well the direct writes work under
concurrent writers, this change:

- Spawns THREAD_COUNT thread workers to perform uploads

- Adds timing support around the portion of the tests performing the
  writes

- Adds a TOTAL_WRITES target to make it clear how many writes were
  performed in a given test.

stevendanna committed:
Populate node objects with pre-generated node data

stevendanna commented:
To generate the node data I did:

```
ohai > testdata/ohaidata.json
```

Not sure if we want to commit that to the repo or just add something that generates it.

tylercloke commented:
If its in the code we should generate it. Maybe add it to the gemfile and `bundle install ohai > testdata/ohaidata.json` as the first step of `benchmark.sh`?

stevendanna commented on scripts/ec/benchmark/solr.rb:
```diff
@@ -0,0 +1,62 @@
+logger.formatter = proc {|severity, datetime, progname, msg| msg + "\n" }
+
+#
+# Returns an array with how many writes
```
This comment is incorrect now because I changed the function.  It doesn't return an array, it takes an int and returns an int.

tylercloke committed:
Added ohai data generation to run automatically.

joedevivo commented on scripts/ec/test.rb:
```diff
@@ -0,0 +1,19 @@
+org = Chef::Config['chef_server_url'].split("/")[-1]
+created_objects = YAML.load_file("testdata/created-objects.yml")
+
+puts "wtf"
```
have we answered this age old question?

tylercloke replied:
LOL

tylercloke replied:
This file wasn't supposed to be vendored at all :( FeelsBadMan
